plan_delegator/src/plan_delegator.cpp

```diff
@@ -201,6 +207,142 @@ namespace plan_delegator
+        else if(starting_lanelet_index > ending_lanelet_index){
+            lane_change_information.is_right_lane_change = false;
+        }
+        else{
+            RCLCPP_WARN_STREAM(get_logger(), "Lane change starting lanelet and ending lanelet have the same lane index!");
+            throw std::invalid_argument("Lane change starting lanelet and ending lanelet have the same lane index!");
+        }
+
+        return lane_change_information;
+    }
+
+    void PlanDelegator::publishUpcomingLaneChangeStatus(const boost::optional<LaneChangeInformation>& upcoming_lane_change_information){
+        // Initialize an UpcomingLaneChangeStatus message, which will be populated based on upcoming_lane_change_information
+        carma_planning_msgs::msg::UpcomingLaneChangeStatus upcoming_lane_change_status;
+
+        // Update upcoming_lane_change_status
+        if(upcoming_lane_change_information){
+            // Get the downtrack distance between the ego vehicle and the start of the upcoming lane change maneuver
+            lanelet::BasicPoint2d current_loc(latest_pose_.pose.position.x, latest_pose_.pose.position.y);
+            double current_downtrack = wm_->routeTrackPos(current_loc).downtrack;       
+            upcoming_lane_change_status.downtrack_until_lanechange = std::max(0.0, upcoming_lane_change_information.get().starting_downtrack - current_downtrack);
+
+            // Set upcoming lane change status as a right lane change or left lane change
+            if(upcoming_lane_change_information.get().is_right_lane_change){
+                upcoming_lane_change_status.lane_change = carma_planning_msgs::msg::UpcomingLaneChangeStatus::RIGHT;
+            }
+            else{
+                upcoming_lane_change_status.lane_change = carma_planning_msgs::msg::UpcomingLaneChangeStatus::LEFT;
+            }
+        }
+        else{
+            upcoming_lane_change_status.lane_change = carma_planning_msgs::msg::UpcomingLaneChangeStatus::NONE;
+        }
+
+        // Publish upcoming_lane_change_status
+        upcoming_lane_change_status_pub_->publish(upcoming_lane_change_status);
+
+        // Store UpcomingLaneChangeStatus in upcoming_lane_change_status_
+        upcoming_lane_change_status_ = upcoming_lane_change_status;
+    }
+
+    void PlanDelegator::publishTurnSignalCommand(const boost::optional<LaneChangeInformation>& current_lane_change_information, const carma_planning_msgs::msg::UpcomingLaneChangeStatus& upcoming_lane_change_status)
+    {
+        // Initialize turn signal command message
+        autoware_msgs::msg::LampCmd turn_signal_command;
+
+        // Publish turn signal command with priority placed on the current lane change, if one exists
+        if(current_lane_change_information){
+            // Publish turn signal command for the current lane change based on the lane change direction
+            if(current_lane_change_information.get().is_right_lane_change){
+                turn_signal_command.r = 1;
```

Comment:
Are these booleans? Can we use true/false instead of 1/0?

Reply:
These are Int32 fields as specified in the autoware_msgs/LampCmd.msg [(Message Definition Here)](https://github.com/streetdrone-home/Autoware/blob/master/ros/src/msgs/autoware_msgs/msg/LampCmd.msg). Additionally, the fields are processed in twist_gate [here](https://github.com/usdot-fhwa-stol/autoware.ai/blob/209ab19703948b854eb68473b46625e8941e6086/drivers/as/nodes/ssc_interface/ssc_interface.cpp#L301). I will add a note in the code to specify this.